### PR TITLE
Don't ignore missing stubs in setuptools

### PR DESCRIPTION
**Pull request by Avasam**
https://github.com/python/typeshed/pull/10057#issuecomment-1511937402
> Given the push for the deprecation of distutils, it'd be a good idea indeed to ensure that all stubs in setuptools are accounted for.

- Fixed stubtest_allowlist entries
- Removed `ignore_missing_stub = true`
- Added missing `__all__`
- Added new default values
- Standardized imports (there was a mix of relative and absolute)

### CODE REVIEW

File: stubs/setuptools/setuptools/depends.pyi [hunk 1]
```diff
@@ -1,4 +1,10 @@
 from _typeshed import Incomplete
+from typing import IO, Any
+from typing_extensions import Literal
+
+__all__ = ["Require", "find_module", "get_module_constant", "extract_constant"]
+
+def find_module(module, paths=None) -> tuple[IO[Any], str, tuple[str, Literal["", "r", "rb"], Literal[7, 6, 1, 2, 3, -1]]]: ...
```
Comment (JelleZijlstra) [inline stubs/setuptools/setuptools/depends.pyi:7]:
```suggestion
def find_module(module, paths=None) -> tuple[IO[Any], str | None, tuple[str, Literal["", "r", "rb"], Literal[7, 6, 1, 2, 3, -1]]]: ...
```
I see some paths where it's None.

**Review by AlexWaygood — COMMENTED** [review]
I haven't finished looking at this, but here's a partial review I started -- mostly a few suggestions for some type annotations that look fairly obvious :)

File: stubs/setuptools/pkg_resources/__init__.pyi [hunk 2]
```diff
@@ -191,8 +213,8 @@ def resource_listdir(package_or_requirement: _PkgReqType, resource_name: str) ->
 def resource_filename(package_or_requirement: _PkgReqType, resource_name: str) -> str: ...
 def set_extraction_path(path: str) -> None: ...
 def cleanup_resources(force: bool = False) -> list[str]: ...
-
-class IResourceManager:
+@type_check_only
+class IResourceManager(Protocol):
```
[the remark this answers is not in the document]
Comment (Avasam) [reply stubs/setuptools/pkg_resources/__init__.pyi:217]: Isn't that what `@type_check_only` is for? Granted support is still limited in mypy (https://github.com/python/mypy/issues/15146, https://github.com/python/mypy/issues/9531)

The source docstrings do call it `IResourceManager`, so it's the right name.

[the remark this answers is not in the document]
Comment (AlexWaygood) [reply stubs/setuptools/pkg_resources/__init__.pyi:217]: True, mypy has been pretty slow to implement this feature :)

Elsewhere in typeshed, though, we generally don't _rely_ on a feature unless it's supported by all major type checkers. We use `@type_check_only` in a _few_ places already, but those are basically all very special cases (`builtins.pyi`, `typing.pyi`) where our standard option of making the name private isn't available.

File: stubs/setuptools/setuptools/command/editable_wheel.pyi [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    mode: Incomplete
+    def initialize_options(self) -> None: ...
+    package_dir: Incomplete
+    def finalize_options(self) -> None: ...
+    def run(self) -> None: ...
+
+class EditableStrategy(Protocol):
+    def __call__(self, wheel: _WheelFile, files: list[str], mapping: dict[str, str]): ...
+    def __enter__(self) -> None: ...
+    def __exit__(self, _exc_type, _exc_value, _traceback) -> None: ...
```
[the remark this answers is not in the document]
Comment (Avasam) [reply stubs/setuptools/setuptools/command/editable_wheel.pyi:36]: Is it fine if all setuptool classes that implement the `EditableStrategy` protocol do return None?

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stubs/setuptools/setuptools/command/editable_wheel.pyi:36]: Maybe for `__exit__`, but it feels too restrictive for `__enter__`.

**Comment by Avasam** [comment]
Oh neat, you can request review from more than one person now on GitHub.

**Comment by Avasam** [comment]
Pretty sure pyright is failing here because of `types-setuptools` or something like that. (notice there's no error for `FileMetadata`, `PathMetadata` and `EggMetadata`). Same errors locally with `types-setuptools` installed.

**Comment by AlexWaygood** [comment]
> Pretty sure pyright is failing here because `types-setuptools` is installed (https://github.com/python/typeshed/actions/runs/4839817600/jobs/8625083762?pr=10058#step:4:55). (notice there's no error for `FileMetadata`, `PathMetadata` and `EggMetadata`). Same errors locally with `types-setuptools` installed.

Seems your diagnosis was spot on!

**Review by JelleZijlstra — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Looks good, @AlexWaygood any further feedback?

@Avasam feel free to mark conversations as "resolved" if you have addressed them.

**Review by AlexWaygood — APPROVED** [review]
Haven't reviewed in depth, but all my previous points have been addressed, nothing stands out as weird from a skim-read, and I trust @JelleZijlstra to do a thorough review :!

**Review by AlexWaygood — APPROVED** [review]
Haven't reviewed in depth, but all my previous points have been addressed, nothing stands out as weird from a skim-read, and I trust @JelleZijlstra to do a thorough review :!

**Pull request merged**